generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -246,11 +238,41 @@ public OpaqueExpressionGenerator(IRandom generator, GenerationParams generationP
                           makeOpaqueOne(type, constContext, depth, fuzzer),
                           shiftValue.clone(),
                           BinOp.SHL)),
-                  shiftValue.clone(),
+                  shiftBackValue,
                   BinOp.SHR)));
     }
   }
 
+  /**
+   * Utility function to clamp a fuzzed expression between an opaque zero and the identity of a
+   * type constructor of the given type, with the value of the bound argument. Note that this
+   * function only supports integer types currently - it could be extended to support floating
+   * point numbers as well if needed. Another note is that this function does not check its bound
+   * for validity - specifying a bound larger than 256 or a negative value could cause invalid GLSL
+   * to be generated depending on the precision and type of the integer.
+   *
+   * @param type - the type to make a clamped expression from.
+   * @param bound - the bound for the randomly generated number.
+   * @return an expression of a clamped value between 0 and m, inclusive:
```

Comment:
I guess you mean "bound" rather than "maxValue" below?

Reply:
Oops!